packages/@react-aria/datepicker/src/useDatePickerGroup.ts

```diff
@@ -9,6 +9,8 @@ import {usePress} from '@react-aria/interactions';
 export function useDatePickerGroup(state: DatePickerState | DateRangePickerState | DateFieldState, ref: RefObject<Element | null>, disableArrowNavigation?: boolean) {
   let {direction} = useLocale();
   let focusManager = useMemo(() => createFocusManager(ref), [ref]);
+  let editableSegments: NodeListOf<Element> | undefined = ref.current?.querySelectorAll('span[role="spinbutton"], span[role="textbox"]');
```

Comment:
this needs to be done in a useLayoutEffect and the result should be stored in a ref as other parts of state will determine if the effects needs to update